dotcom-rendering/src/components/Discussion/Comments.tsx

```diff
@@ -214,6 +220,15 @@ export const Comments = ({
 		dispatch({ type: 'addReply', comment });
 	};
 
+	useEffect(() => {
+		if (isWeb && expanded && !loading && enableMobileDiscussionAdsSwitch) {
+			const event = new CustomEvent('comments-loaded');
+			document.dispatchEvent(event);
+		}
+	}, [isWeb, expanded, loading, enableMobileDiscussionAdsSwitch]);
+
+	const commentsStateChangeEvent = new CustomEvent('comments-state-change');
```

Comment:
Does this need to be defined within the component’s body?

Reply:
A foreboding comment!

Reply:
We should add a test for `DiscussionWeb` in [`Island.test.tsx`](https://github.com/guardian/dotcom-rendering/blob/ae2ee705e9cec5c357e6751427ead2aec16fbcfe/dotcom-rendering/src/components/Island.test.tsx#L19)